test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -480,3 +480,237 @@ def test_res_size_runtime(self):
+        restype = ["string"]
+
+        # Script returns integer value for resource type 'string'
+        script_body = "echo 100"
+
+        fn = self.du.create_temp_file(prefix="PtlPbs_string",
+                                      suffix=".scr",
+                                      body=script_body)
+        self.du.chmod(path=fn, mode=0755, sudo=True)
+
+        resval = ['"' + resname[0] + ' ' + '!' + fn + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Submit job
+        a = {'Resource_List.foo': 'a'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foo (a != 100)"
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_string_invalid_input2(self):
+        """
+        Test invalid values returned from server_dyn_resource
+        script for resource type 'string'
+        Script returns a 'float' instead of type 'string'.
+        """
+        # Create a resource of type string
+        resname = ["foo"]
+        restype = ["string"]
+
+        # Script returns float value for resource type 'string'
+        script_body = "echo 1.25"
+
+        fn = self.du.create_temp_file(prefix="PtlPbs_string",
+                                      suffix=".scr",
+                                      body=script_body)
+        self.du.chmod(path=fn, mode=0755, sudo=True)
+
+        resval = ['"' + resname[0] + ' ' + '!' + fn + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Submit job
+        a = {'Resource_List.foo': 'abc'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foo (abc != 1.25)"
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_string_array_invalid_input1(self):
+        """
+        Test invalid values returned from server_dyn_resource
+        script for resource type 'string_array'
+        Script returns a 'string' instead of type 'string_array'.
+        """
+        # Create a resource of type string_array
+        resname = ["foo"]
+        restype = ["string_array"]
+
+        # Prep for server_dyn_resource script
+        script_body = "echo red blue"
+
+        fn = self.du.create_temp_file(prefix="PtlPbs_str_arr",
+                                      suffix=".scr",
+                                      body=script_body)
+        self.du.chmod(path=fn, mode=0755, sudo=True)
+
+        resval = ['"' + resname[0] + ' ' + '!' + fn + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Submit job
+        a = {'Resource_List.foo': '"red,blue"'}
```

Comment:
String arrays are a one to N match.  You don't request "red,blue", you request "red" or "blue".  You can't request multiple values of a string array.  Your string array has one value in it ("red blue").  You are requesting the string "red,blue" and it does not match the one value "red blue".  You aren't really testing anything special here.